core/src/apps/management/authenticate_device.py

```diff
@@ -0,0 +1,46 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from trezor.messages import AuthenticateDevice, AuthenticityProof
+
+
+async def authenticate_device(msg: AuthenticateDevice) -> AuthenticityProof:
+    from trezor import utils
+    from trezor.crypto import optiga
+    from trezor.crypto.der import read_length
+    from trezor.crypto.hashlib import sha256
+    from trezor.messages import AuthenticityProof
+    from trezor.ui.layouts import confirm_action
+    from trezor.utils import BufferReader
+
+    from apps.common.writers import write_compact_size
+
+    await confirm_action(
+        "authenticate_device",
+        "Authenticate device",
+        description="Do you wish to verify the authenticity of your device?",
+    )
+
+    header = b"AuthenticateDevice:"
+    h = utils.HashWriter(sha256())
+    write_compact_size(h, len(header))
+    h.extend(header)
+    write_compact_size(h, len(msg.challenge))
+    h.extend(msg.challenge)
+    signature = optiga.sign(optiga.DEVICE_ECC_KEY_INDEX, h.get_digest())
```

Comment:
this should catch `SigningInaccessible` and convert to, probably, `ProcessError`, talking about unlocked bootloader? maybe even an error pop-up?

Reply:
Partially done in https://github.com/trezor/trezor-firmware/pull/3255/commits/6928d6f24434a5e5721b608de059eedde7f356e0. I wonder whether we shouldn't provide an earlier indication that the bootloader is unlocked, e.g. in `Features` or respond with an error before the user is asked to confirm.

Reply:
it should probably be part of Features, https://github.com/trezor/trezor-firmware/issues/3268